src/content/insertable-streams/endtoend-encryption/js/videopipe.js

```diff
@@ -36,36 +36,37 @@ function VideoPipe(stream, forceSend, forceReceive, handler) {
   this.pc2 = new RTCPeerConnection({
     encodedInsertableStreams: forceReceive,
   });
-  this.pc2.ontrack = handler;
-  stream.getTracks().forEach((track) => this.pc1.addTrack(track, stream));
-
-  if (preferredAudioCodecMimeType && 'setCodecPreferences' in window.RTCRtpTransceiver.prototype) {
-    const {codecs} = RTCRtpSender.getCapabilities('audio');
-    const selectedCodecIndex = codecs.findIndex(c => c.mimeType === preferredAudioCodecMimeType);
-    const selectedCodec = codecs[selectedCodecIndex];
-    codecs.splice(selectedCodecIndex, 1);
-    codecs.unshift(selectedCodec);
-    const transceiver = this.pc1.getTransceivers()
-        .find(t => t.sender && t.sender.track === stream.getAudioTracks()[0]);
-    transceiver.setCodecPreferences(codecs);
-  }
-  if (preferredVideoCodecMimeType && 'setCodecPreferences' in window.RTCRtpTransceiver.prototype) {
-    const {codecs} = RTCRtpSender.getCapabilities('video');
-    const selectedCodecIndex = codecs.findIndex(c => c.mimeType === preferredVideoCodecMimeType);
-    const selectedCodec = codecs[selectedCodecIndex];
-    codecs.splice(selectedCodecIndex, 1);
-    codecs.unshift(selectedCodec);
-    const transceiver = this.pc1.getTransceivers()
-        .find(t => t.sender && t.sender.track === stream.getVideoTracks()[0]);
-    transceiver.setCodecPreferences(codecs);
+  if ('setCodecPreferences' in window.RTCRtpTransceiver.prototype) {
+    this.pc2.ontrack = (e) => {
+      if (e.track.kind === 'audio' && preferredAudioCodecMimeType ) {
```

Comment:
Hm. From surrounding text, it looks as if this should be a handler function, not an innate property of VideoPipe.
But that's a different refactoring from this one, so not a showstopper.

Reply:
The good news is that we can now actually move it to the outside ontrack handler